src/Formik.tsx

```diff
@@ -118,28 +122,53 @@ export class Formik<ExtraProps = {}, Values = object> extends React.Component<
     }
   }
 
+  internalSetState = (
+    stateToSet:
+      | Pick<FormikState<any>, never>
+      | ((state: Readonly<FormikState<any>>) => any),
+    cb?: () => any
+  ) => {
+    if (!this.didMount) {
+      return undefined;
+    }
+
+    return this.setState(
+      state => {
+        let newStateToSet =
+          typeof stateToSet === 'function' ? stateToSet(state) : stateToSet;
+        newStateToSet = this.props.stateReducer!(state, newStateToSet);
```

Comment:
The exclamation mark: is it a TypeScript thing?

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Since it's an optional property, it will show an error that it might be undefined, but it won't be because of `defaultProps`